### ISSUE TITLE
[T-SQL] NO ACTION not accepted as argument for ON DELETE

### ISSUE THREAD

**Issue by flipflop97** [sqlfluff/sqlfluff#5432]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

NO ACTION is not accepted as argument for ON DELETE when creating or altering a table with a constraint.

### Expected Behaviour

NO ACTION should be accepted as argument (next to CASCADE, SET NULL or SET DEFAULT, which are accepted).

https://learn.microsoft.com/en-us/sql/t-sql/statements/alter-table-column-constraint-transact-sql?view=sql-server-ver16&redirectedfrom=MSDN#syntax

### Observed Behaviour

```
WARNING    Unable to lint test.sql due to an internal error. Please report this as an issue with your query's contents and stacktrace below!                                                                                                         
To hide this warning, add the failing file to .sqlfluffignore
RuntimeError: Grammar refers to the 'Action' keyword which was not found in the tsql dialect.

The syntax in the query is not (yet?) supported. Try to narrow down your query to a minimal, reproducible case and raise an issue on GitHub.

Or, even better, see this guide on how to help contribute keyword and/or dialect updates:
https://github.com/sqlfluff/sqlfluff/wiki/Contributing-Dialect-Changes#keywords
 
All Finished 📜 🎉!
```

### How to reproduce

```bash
sqlfluff lint test.sql --dialect tsql
```

Where test.sql contains

```SQL
CREATE TABLE table_name (
    id UNIQUEIDENTIFIER NOT NULL
    CONSTRAINT constraint_name
    REFERENCES referenced_table_name
    ON DELETE NO ACTION
);
```

### Dialect

Transact-SQL

### Version

- sqlfluff, version 2.3.5
- Python 3.11.6

### Configuration

[sqlfluff]
dialect = tsql

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
